lib/Demangling/Demangler.cpp

```diff
@@ -783,7 +782,22 @@ NodePointer Demangler::demangleOperator() {
     case 'V': return demangleAnyGenericType(Node::Kind::Structure);
     case 'W': return demangleWitness();
     case 'X': return demangleSpecialType();
-    case 'Y': return createNode(Node::Kind::AsyncAnnotation);
+    case 'Y':
+      switch (char c2 = nextChar()) {
```

Comment:
cosmetic: Can you extract this switch to a separate function? Just to keep ``demangleOperator`` readable.

Reply:
Extracted to `Demangler::demangleTypeAnnotation()`.